subsys/net/ip/ipv6.c

```diff
@@ -1432,6 +1432,12 @@ static enum net_verdict handle_ns_input(struct net_pkt *pkt)
 				goto drop;
 			}
 
+			if (nd_opt_hdr->len > 2) {
+				NET_ERR("Invalid source link-layer address "
```

Comment:
I would not call this "Invalid ... length", it is just that the we do not support such long link layer address in zephyr currently. So we could say here "Too long source link-layer address length ...."
Same thing for commit message, this issue is in ipv6 and not in icmpv6, so change it to "net: ipv6: Drop NS packet if LL address len is too long"